src/platform/vita/vita_touch.c

```diff
@@ -0,0 +1,343 @@
+
+#include "vita_touch.h"
+
+#include "math.h"
+
+static void init_touch(void);
+static void preprocess_events(SDL_Event *event);
+static void preprocess_finger_down(SDL_Event *event);
+static void preprocess_finger_up(SDL_Event *event);
+static void preprocess_finger_motion(SDL_Event *event);
+
+static int vita_rear_touch = 0; // always disable rear_touch for now
+extern int last_mouse_x;
+extern int last_mouse_y;
+
+static int touch_initialized = 0;
+static unsigned int simulated_click_start_time[SCE_TOUCH_PORT_MAX_NUM][2]; // initiation time of last simulated left or right click (zero if no click)
+
+enum {
+    MAX_NUM_FINGERS = 3, // number of fingers to track per panel
+    MAX_TAP_TIME = 250, // taps longer than this will not result in mouse click events
+    MAX_TAP_MOTION_DISTANCE = 10, // max distance finger motion in Vita screen pixels to be considered a tap
+    SIMULATED_CLICK_DURATION = 50, // time in ms how long simulated mouse clicks should be
+}; // track three fingers per panel
+
+typedef struct {
+    int id; // -1: no touch
+    Uint32 time_last_down;
+    int last_x; // last known screen coordinates
+    int last_y; // last known screen coordinates
+    float last_down_x; // SDL touch coordinates when last pressed down
+    float last_down_y; // SDL touch coordinates when last pressed down
+} Touch;
+
+static Touch finger[SCE_TOUCH_PORT_MAX_NUM][MAX_NUM_FINGERS]; // keep track of finger status
+
+typedef enum DraggingType {
+    DRAG_NONE = 0,
+    DRAG_TWO_FINGER,
+    DRAG_THREE_FINGER,
+} DraggingType;
+
+static DraggingType multi_finger_dragging[SCE_TOUCH_PORT_MAX_NUM]; // keep track whether we are currently drag-and-dropping
+
+typedef int bool;
+#define false 0;
+#define true 1;
+
+static void init_touch(void)
+{
+    for (int port = 0; port < SCE_TOUCH_PORT_MAX_NUM; port++) {
+        for (int i = 0; i < MAX_NUM_FINGERS; i++) {
+            finger[port][i].id = -1;
+        }
+        multi_finger_dragging[port] = DRAG_NONE;
+    }
+
+    for (int port = 0; port < SCE_TOUCH_PORT_MAX_NUM; port++) {
+        for (int i = 0; i < 2; i++) {
+            simulated_click_start_time[port][i] = 0;
+        }
+    }
+}
+
+void vita_handle_touch(SDL_Event *event)
+{
+    if (!touch_initialized) {
+        init_touch();
+        touch_initialized = 1;
+    }
+    preprocess_events(event);
+}
+
+static void preprocess_events(SDL_Event *event)
+{
+    // prevent suspend
+    sceKernelPowerTick(SCE_KERNEL_POWER_TICK_DISABLE_AUTO_SUSPEND);
+    sceKernelPowerTick(SCE_KERNEL_POWER_TICK_DISABLE_OLED_OFF);
+
+    // Supported touch gestures:
+    // left mouse click: single finger short tap
+    // right mouse click: second finger short tap while first finger is still down
+    // pointer motion: single finger drag
+    // left button drag and drop: dual finger drag
+    // right button drag and drop: triple finger drag
+    if (event->type == SDL_FINGERDOWN || event->type == SDL_FINGERUP || event->type == SDL_FINGERMOTION) {
```

Comment:
```suggestion
    if (event->type != SDL_FINGERDOWN && event->type != SDL_FINGERUP && event->type != SDL_FINGERMOTION)
        return;
```

Please do this for every conditional enclosing all the statements.

Reply:
Fixed in 35f42ab00e2b28a49e966a752eb3016a334d53bb